test/SynchroMenuTest.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Combodo\iTop\Test\SynchroDashboard;
+
+use ApplicationMenu;
+use Combodo\iTop\Test\UnitTest\ItopDataTestCase;
+use MetaModel;
+use SynchroDataSource;
+use UserRights;
+
+
+/**
+ * @runTestsInSeparateProcesses
+ * @backupGlobals disabled
+ */
+class SynchroMenuTest extends ItopDataTestCase {
+	protected function setUp(): void {
+		parent::setUp();
+	}
+
+	public function MenuAccessProvider(){
+		return [
+			'Administrator' => [ 'sProfileName' => 'Administrator' ],
+			'Configuration Manager' => [ 'sProfileName' => 'Configuration Manager' ],
+			'Support Agent' => [ 'sProfileName' => 'Support Agent' ],
+			/*'ProfileWithSynchroData' => [ 'sProfileName' => 'ProfileWithSynchroData' ],
+			'ProfileWithAdminTools' => [ 'sProfileName' => 'ProfileWithAdminTools' ],
+			'ProfileWithoutSynchro' => [ 'sProfileName' => 'ProfileWithoutSynchro' ],*/
+		];
+	}
+
+	/*private static function CleanupEnv(string $sEnv){
+		$aFilesToCleanup = [
+			\utils::GetConfigFilePath($sEnv),
+			APPROOT . "/conf/$sEnv",
+			APPROOT . "/data/cache-$sEnv",
+			APPROOT . "/data/env-$sEnv",
+			APPROOT . "/data/datamodel-$sEnv-with-delta.xml",
+			APPROOT . "/data/datamodel-$sEnv.xml",
+			APPROOT . "/data/$sEnv.delta.xml",
+		];
+
+		foreach ($aFilesToCleanup as $sPath){
+			if (is_file($sPath)){
+				@unlink($sPath);
+			}
+
+			if (is_dir($sPath)){
+				self::CleanupDir($sPath);
+			}
+		}
+	}
+
+	private static function CleanupDir(string $sDir){
+		if (! is_dir($sDir)){
+			return;
+		}
+
+		foreach (glob("$sDir" . DIRECTORY_SEPARATOR . '*') as $sPath){
+			if (is_dir($sPath)){
+				self::CleanupDir($sPath);
+			}
+
+			if (is_file($sPath)){
+				@unlink($sPath);
+			}
+		}
+
+		@rmdir($sDir);
+	}
+
+	private function CompileDeltaIfRequired(string $sEnv){
+		$sConfigFilePath = \utils::GetConfigFilePath($sEnv);
+
+		if (! is_file($sConfigFilePath)) {
+			copy(__DIR__. "/production.delta.xml", APPROOT . "/data/$sEnv.delta.xml");
+
+			//copy conf from production to phpunit context
+			$sDirPath = dirname($sConfigFilePath);
+			if (!is_dir($sDirPath)) {
+				mkdir($sDirPath);
+			}
+			$oConfig = new Config(\utils::GetConfigFilePath());
+			$oConfig->WriteToFile($sConfigFilePath);
+
+			$oConfig = new Config($sConfigFilePath);
```

Comment:
Can we clean all of this?

Reply:
i would like to keep it and uncomment this section/adapt it to use ongoing PR regarding datamodels.... this could be our lab to validate the PR :)